Sources/SourceKitLSP/Swift/MacroExpansion.swift

```diff
@@ -111,22 +119,96 @@ extension SwiftLanguageService {
           )
         }
 
-        Task {
-          let req = ShowDocumentRequest(uri: DocumentURI(macroExpansionFilePath), selection: macroEdit.range)
+        macroExpansionFilePaths.append(macroExpansionFilePath)
 
-          let response = await orLog("Sending ShowDocumentRequest to Client") {
-            try await sourceKitLSPServer.sendRequestToClient(req)
-          }
+        let editContent =
+          """
+          // \(sourceFileURL.lastPathComponent) @ \(macroEdit.range.lowerBound.line + 1):\(macroEdit.range.lowerBound.utf16index + 1) - \(macroEdit.range.upperBound.line + 1):\(macroEdit.range.upperBound.utf16index + 1)
+          \(macroEdit.newText)
 
-          if let response, !response.success {
-            logger.error("client refused to show document for \(expansion.title, privacy: .public)")
-          }
-        }
+
+          """
+        completeExpansionFileContent += editContent
       } else if !macroEdit.newText.isEmpty {
         logger.fault("Unable to retrieve some parts of macro expansion")
       }
     }
 
-    return expansion.edits.encodeToLSPAny()
+    // removes superfluous newline
+    if completeExpansionFileContent.hasSuffix("\n\n") {
+      completeExpansionFileContent.removeLast()
+    }
+
+    if completeExpansionDirectoryName.hasSuffix("-") {
+      completeExpansionDirectoryName.removeLast()
+    }
+
+    var completeExpansionFilePath =
+      self.generatedMacroExpansionsPath.appendingPathComponent(
+        Insecure.MD5.hash(
+          data: Data(completeExpansionDirectoryName.utf8)
+        )
+        .map { String(format: "%02hhx", $0) }
+        .joined()
```

Comment:
What kind of directory name does this produce? How does this compare to the `.description` that we do in `hashForLogging`?

Reply:
I mistook `.description` to provide some debugging description. Seems its the same as what I am trying to achieve. Will change this. 👍🏻

Reply:
After playing around with it, seems I was right. `.description` indeed provides a debug description "MD5 Digest: e1c37d2fd6a2a2efdfab947c8715b5e4".
My original method, maps each byte into hex and joins them into a string. and it outputs the following "e1c37d2fd6a2a2efdfab947c8715b5e4"

I guess my method is fine.

Reply:
Oh, interesting. Thanks for adding the comment.